src/main/java/org/junit/internal/Checks.java

```diff
@@ -0,0 +1,37 @@
+package org.junit.internal;
+
+/** @since 4.13 */
+public final class Checks {
+
+    private Checks() {}
+
+    /**
+     * Checks that the given value is not {@code null}.
+     *
+     * @param value object reference to check
+     * @return the passed-in value, if not {@code null}
+     * @throws NullPointerException if {@code value} is {@code null}
+     */
+    public static <T> T notNull(T value) {
+        if (value == null) {
+            throw new NullPointerException();
+        }
+        return value;
+    }
+
+    /**
+     * Checks that the given value is not {@code null}, using the given message
+     * as the exception message if an exception is thrown.
+     *
+     * @param value object reference to check
+     * @param message message to use if {@code value} is {@code null}
+     * @return the passed-in value, if not {@code null}
+     * @throws NullPointerException if {@code value} is {@code null}
+     */
+    public static <T> T notNull(T value, String message) {
+        if (value == null) {
+            throw new NullPointerException(String.valueOf(message));
```

Comment:
The behavior I have here matches Guava's `Preconditions.checkNotNull(T, String)` method. I think the Guava team has given this much more thought than I have, but I can take a stab at explaining why the behavior I have is better.

IMHO it is a programmer error to pass in a `null` as the second parameter. If we have this method behave like `notNull(T)` when you pass in a `null` message we would be ignoring that programmer error. Having the message be `"null"` makes it clear that 1) someone tried to specify a real message, and 2) they messed up. Hopefully someone would notice and fix it.